From f3308f1b663af916d101d429331295e1b74e76e8 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 17 Feb 2021 13:47:46 -0700
Subject: [PATCH 17/18] DO-NOT-UPSTREAM: Various fixes/hacks/warning hiding

* drivers/input/joystick/analog.c: Avoid this warning on s390.

* lib/{Kconfig.debug,Makefile}: Disable -Wframe-larger-than=. Getting
  all of these fixed is a long term project.

* scripts/checksyscalls.sh: I do not care for this warning.

Link: https://github.com/ClangBuiltLinux/linux/labels/-Wframe-larger-than%3D
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/input/joystick/analog.c | 1 -
 lib/Kconfig.debug               | 3 ++-
 lib/Makefile                    | 1 -
 scripts/checksyscalls.sh        | 2 +-
 4 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/input/joystick/analog.c b/drivers/input/joystick/analog.c
index f798922a4598..dea166758af1 100644
--- a/drivers/input/joystick/analog.c
+++ b/drivers/input/joystick/analog.c
@@ -157,7 +157,6 @@ static unsigned long analog_faketime = 0;
 #define GET_TIME(x)     do { x = analog_faketime++; } while(0)
 #define DELTA(x,y)	((y)-(x))
 #define TIME_NAME	"Unreliable"
-#warning Precise timer not defined for this architecture.
 #endif
 
 static inline u64 get_time(void)
diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 5ddd575159fb..e299f71a48af 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -343,8 +343,9 @@ config GDB_SCRIPTS
 endif # DEBUG_INFO
 
 config FRAME_WARN
-	int "Warn for stack frames larger than"
+	int "Warn for stack frames larger than" if !CC_IS_CLANG
 	range 0 8192
+	default 0 if CC_IS_CLANG
 	default 2048 if GCC_PLUGIN_LATENT_ENTROPY
 	default 1280 if (!64BIT && PARISC)
 	default 1024 if (!64BIT && !PARISC)
diff --git a/lib/Makefile b/lib/Makefile
index 5efd1b435a37..eea1027b4206 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -351,7 +351,6 @@ obj-$(CONFIG_OBJAGG) += objagg.o
 obj-$(CONFIG_PLDMFW) += pldmfw/
 
 # KUnit tests
-CFLAGS_bitfield_kunit.o := $(call cc-option,-Wframe-larger-than=10240)
 obj-$(CONFIG_BITFIELD_KUNIT) += bitfield_kunit.o
 obj-$(CONFIG_LIST_KUNIT_TEST) += list-test.o
 obj-$(CONFIG_LINEAR_RANGES_TEST) += test_linear_ranges.o
diff --git a/scripts/checksyscalls.sh b/scripts/checksyscalls.sh
index b7609958ee36..5f42da1b0a1a 100755
--- a/scripts/checksyscalls.sh
+++ b/scripts/checksyscalls.sh
@@ -259,7 +259,7 @@ EOF
 syscall_list() {
     grep '^[0-9]' "$1" | sort -n |
 	while read nr abi name entry ; do
-		echo "#if !defined(__NR_${name}) && !defined(__IGNORE_${name})"
+		echo "#if 0"
 		echo "#warning syscall ${name} not implemented"
 		echo "#endif"
 	done
-- 
2.33.0

